### PR TITLE
Add an action to lint shell scripts and Dockerfiles

### PR DESCRIPTION
**Pull request by zb140**
Added an action to run linters for shell scripts and Dockerfiles.  It will run on any PR.  If there's a lint issue in one of the files being modified, it will add a review pointing out the specific line and error. If it finds errors in files _not_ in the PR, those will get noted under the "Checks" tab.  None of this blocks merging the PR anyway, but it should make it much easier to spot potential problems.

### CODE REVIEW

File: run-gow [hunk 1]
```diff
@@ -254,7 +254,7 @@ function os_type() {
         os="unraid"
     elif [ -f /etc/lsb-release ]; then
         # shellcheck disable=1091
-        os="$(source /etc/lsb-release; echo "$DISTRIB_ID")"
+        os= "$(source /etc/lsb-release; echo "$DISTRIB_ID")"
```
[the remark this answers is not in the document]
Comment (zb140) [reply run-gow]: I have fixed this but not marked it "Resolved" just so that it can be an example of what a lint error would look like.

**Comment by zb140** [comment]
Note that I have not yet gone through and fixed all the existing lint issues; my plan is to do that after merging #76, since there's changes to pretty much all the files that would be affected by this.

**Comment by Sparticuz** [comment]
While we're at it...

https://github.com/dotenv-linter/action-dotenv-linter

https://github.com/marketplace/actions/yaml-lint

**Comment by zb140** [comment]
> While we're at it...
> 
> https://github.com/dotenv-linter/action-dotenv-linter
> 
> https://github.com/marketplace/actions/yaml-lint

Good idea! I've added them.

**Comment by Sparticuz** [comment]
Lgtm

**Pull request merged**